kotest-assertions/kotest-assertions-yaml/src/commonMain/kotlin/io/kotest/assertions/yaml/YamlMatchers.kt

```diff
@@ -58,8 +60,25 @@ fun equalYaml(
             { "Expected value to be not equal to YAML '$expected', but was: null" }
          )
       } else {
-         val (expectedTree, actualTree) = parse(expected, actual)
-         equalYamlNode(expectedTree).test(actualTree)
+         val actualYaml = try {
+            yaml.parseToYamlNode(actual)
+         } catch (ex: Exception) {
+            return@Matcher MatcherResult(
+               false,
+               { "expected: actual YAML to be valid YAML Reason: ${ex.message} Actual: $actual" },
+               { "expected: actual YAML to be invalid YAML: $actual" }
+            )
+         }
+         val expectedYaml = try {
+            yaml.parseToYamlNode(expected)
+         } catch (ex: Exception) {
+            return@Matcher MatcherResult(
+               false,
+               { "expected: actual YAML to be valid YAML Reason: ${ex.message} Actual: $actual" },
+               { "expected: expected YAML to be invalid YAML: $expected" }
+            )
+         }
+         equalYamlNode(expectedYaml).test(actualYaml)
```

Comment:
I'm wondering if these exceptions should instead be re-thrown.

If we are passing something that's not an YAML, we are not making an assertion, we are failing anyways. Having a hard failure for bad yamls would clean up a bit these lines

Reply:
I understand your reasoning. I was following the JSON matchers here. They have the same behaviour. In case of an invalid JSON they catch the exception and return the failed result.